File: dataset_reduzido/codenet/Java/Code/s918182055.java
import java.util.*;
import java.io.*;
import java.awt.Point;

public class Main {
  
    public static void main(String[] args) {
      MyWriter out = new MyWriter(System.out);
      
      MyScanner sc = new MyScanner(System.in);
      int N = sc.nextInt();
      int M = sc.nextInt();
      LinkedList<Integer>[] list = new LinkedList[100001];

      for (int i = 0; i < N; i++) {
        int a = sc.nextInt();
      	int b = sc.nextInt();
        if (list[a] == null) {
          list[a] = new LinkedList<Integer>();
        }
        list[a].add(b);
      }
      
      int res = 0;
      PriorityQueue <Integer> pqueue = new PriorityQueue<Integer>(Collections.reverseOrder());
      for (int i = 1; i <= M; i++) {
        if (list[i] != null) {
          pqueue.addAll(list[i]);
        }
        if (!pqueue.isEmpty())
        	res += pqueue.poll();
      }
      
      out.println(res);
      out.flush();
    }
  
    static final class MyScanner {
        static final int BUFFER_SIZE = 8192;
        private final InputStream in;
        private final byte[] buffer = new byte[BUFFER_SIZE];
        private int point;
        private int readLength;
        MyScanner(InputStream in) {
            this.in = in;
        }
        private int readByte() {
            if (point < readLength) {
                int result = buffer[point];
                point += 1;
                return result;
            }
            try {
                readLength = in.read(buffer);
            } catch (IOException e) {
                throw new AssertionError(null, e);
            }
            if (readLength == -1) {
                return -1;
            }
            point = 1;
            return buffer[0];
        }
        private static boolean isPrintableCharExceptSpace(int c) {
            return 33 <= c && c <= 126;
        }
        String next() {
            int c = readByte();
            while (!(c == -1 || isPrintableCharExceptSpace(c))) {
                c = readByte();
            }
            if (c == -1) {
                throw new NoSuchElementException();
            }
            StringBuilder b = new StringBuilder();
            do {
                b.appendCodePoint(c);
                c = readByte();
            } while (c != -1 && isPrintableCharExceptSpace(c));
            return b.toString();
        }
        long nextLong() {
            int c = readByte();
            while (!(c == -1 || isPrintableCharExceptSpace(c))) {
                c = readByte();
            }
            if (c == -1) {
                throw new NoSuchElementException();
            }
            boolean minus = false;
            if (c == '-') {
                minus = true;
                c = readByte();
            }
            long result = 0L;
            do {
                if (c < '0' || c > '9') {
                    throw new InputMismatchException();
                }
                result = result * 10L + (c - '0');
                c = readByte();
            } while (c != -1 && isPrintableCharExceptSpace(c));
            return minus ? -result : result;
        }
        int nextInt() {
            long n = nextLong();
            if (n < Integer.MIN_VALUE || n > Integer.MAX_VALUE) {
                throw new InputMismatchException();
            }
            return (int)n;
        }
        double nextDouble() {
            return Double.parseDouble(next());
        }
        int[] nextIntArray(int n) {
            int[] result = new int[n];
            for (int i = 0; i < n; i++) {
                result[i] = nextInt();
            }
            return result;
        }
        private static boolean allSameLength(int[] a, int[] b, int[]... c) {
            if (a.length != b.length) {
                return false;
            }
            for (int[] element : c) {
                if (a.length != element.length) {
                    return false;
                }
            }
            return true;
        }
        private static boolean allSameLength(char[] a, char[] b, char[]... c) {
            if (a.length != b.length) {
                return false;
            }
            for (char[] element : c) {
                if (a.length != element.length) {
                    return false;
                }
            }
            return true;
        }
        void nextVerticalIntArrays(int[] a, int[] b, int[]... c) {
            if (!allSameLength(a, b, c)) {
                throw new IllegalArgumentException();
            }
            for (int i = 0; i < a.length; i++) {
                a[i] = nextInt();
                b[i] = nextInt();
                for (int[] d : c) {
                    d[i] = nextInt();
                }
            }
        }
        long[] nextLongArray(int n) {
            long[] result = new long[n];
            for (int i = 0; i < n; i++) {
                result[i] = nextLong();
            }
            return result;
        }
        char nextChar() {
            int c = readByte();
            while (!(c == -1 || isPrintableCharExceptSpace(c))) {
                c = readByte();
            }
            if (c == -1) {
                throw new NoSuchElementException();
            }
            return (char)c;
        }
        char[] nextCharArray(int n) {
            char[] result = new char[n];
            for (int i = 0; i < n; i++) {
                result[i] = nextChar();
            }
            return result;
        }
        char[][] next2dCharArray(int n, int m) {
            char[][] result = new char[n][m];
            for (int i = 0; i < n; i++) {
                for (int j = 0; j < m; j++) {
                    result[i][j] = nextChar();
                }
            }
            return result;
        }
        void nextVerticalCharArrays(char[] a, char[] b, char[]... c) {
            if (!allSameLength(a, b, c)) {
                throw new IllegalArgumentException();
            }
            for (int i = 0; i < a.length; i++) {
                a[i] = nextChar();
                b[i] = nextChar();
                for (char[] d : c) {
                    d[i] = nextChar();
                }
            }
        }
    }
    static final class MyWriter extends PrintWriter {
        MyWriter(OutputStream out) {
            super(out);
        }
        void joinAndPrintln(int[] x) {
            joinAndPrintln(x, " ");
        }
        void joinAndPrintln(int[] x, String delimiter) {
            StringBuilder b = new StringBuilder();
            if (x.length > 0) {
                b.append(x[0]);
                for (int i = 1; i < x.length; i++) {
                    b.append(delimiter).append(x[i]);
                }
            }
            println(b.toString());
        }
        void joinAndPrintln(long[] x) {
            joinAndPrintln(x, " ");
        }
        void joinAndPrintln(long[] x, String delimiter) {
            StringBuilder b = new StringBuilder();
            if (x.length > 0) {
                b.append(x[0]);
                for (int i = 1; i < x.length; i++) {
                    b.append(delimiter).append(x[i]);
                }
            }
            println(b.toString());
        }
        void joinAndPrintln(Iterable<?> iterable) {
            joinAndPrintln(iterable, " ");
        }
        void joinAndPrintln(Iterable<?> iterable, String delimiter) {
            StringBuilder b = new StringBuilder();
            for (Iterator<?> i = iterable.iterator(); i.hasNext();) {
                b.append(i.next());
                while (i.hasNext()) {
                    b.append(delimiter).append(i.next());
                }
            }
            println(b.toString());
        }
    }
}

File: output/StarCoder-100/codenet/Java/Python/s918182055.py
import sys

class MyWriter(object):
    def __init__(self, out):
        self.out = out
    def joinAndPrintln(self, x, delimiter=" "):
        b = ""
        if len(x) > 0:
            b += str(x[0])
            for i in range(1, len(x)):
                b += delimiter + str(x[i])
        self.println(b)
    def println(self, s):
        self.out.write(s + "\n")
    def flush(self):
        self.out.flush()

class MyScanner(object):
    def __init__(self, in_):
        self.in_ = in_
        self.buffer = ""
        self.point = 0
    def readByte(self):
        if self.point < len(self.buffer):
            result = self.buffer[self.point]
            self.point += 1
            return result
        self.buffer = self.in_.readline().rstrip()
        self.point = 0
        if self.buffer == "":
            return -1
        return self.buffer[self.point]
    def isPrintableCharExceptSpace(self, c):
        return 33 <= c and c <= 126
    def next(self):
        c = self.readByte()
        while not (c == -1 or self.isPrintableCharExceptSpace(c)):
            c = self.readByte()
        if c == -1:
            raise NoSuchElementException()
        b = ""
        while c!= -1 and self.isPrintableCharExceptSpace(c):
            b += chr(c)
            c = self.readByte()
        return b
    def nextLong(self):
        c = self.readByte()
        while not (c == -1 or self.isPrintableCharExceptSpace(c)):
            c = self.readByte()
        if c == -1:
            raise NoSuchElementException()
        minus = False
        if c == '-':
            minus = True
            c = self.readByte()
        result = 0L
        while c!= -1 and self.isPrintableCharExceptSpace(c):
            if c < '0' or c > '9':
                raise InputMismatchException()
            result = result * 10L + (c - '0')
            c = self.readByte()
        return minus? -result : result
    def nextInt(self):
        n = self.nextLong()
        if n < Integer.MIN_VALUE or n > Integer.MAX_VALUE:
            raise InputMismatchException()
        return (int)n
    def nextDouble(self):
        return Double.parseDouble(self.next())
    def nextIntArray(self, n):
        result = [0] * n
        for i in range(n):
            result[i] = self.nextInt()
        return result
    def nextVerticalIntArrays(self, a, b, *c):
        if not allSameLength(a, b, *c):
            raise IllegalArgumentException()
        for i in range(len(a)):
            a[i] = self.nextInt()
            b[i] = self.nextInt()
            for d in c:
                d[i] = self.nextInt()
    def nextLongArray(self, n):
        result = [0L] * n
        for i in range(n):
            result[i] = self.nextLong()
        return result
    def nextChar(self):
        c = self.readByte()
        while not (c == -1 or self.isPrintableCharExceptSpace(c)):
            c = self.readByte()
        if c == -1:
            raise NoSuchElementException()
        return chr(c)
    def nextCharArray(self, n):
        result = [0] * n
        for i in range(n):
            result[i] = self.nextChar()
        return result
    def next2dCharArray(self, n, m):
        result = [[0] * m] * n
        for i in range(n):
            for j in range(m):
                result[i][j] = self.nextChar()
        return result
    def nextVerticalCharArrays(self, a, b, *c):
        if not allSameLength(a, b, *c):
            raise IllegalArgumentException()
        for i in range(len(a)):
            a[i] = self.nextChar()
            b[i] = self.nextChar()
            for d in c:
                d[i] = self.nextChar()

class NoSuchElementException(Exception):
    pass

class InputMismatchException(Exception):
    pass

class IllegalArgumentException(Exception):
    pass

def allSameLength(a, b, *c):
    if
